src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -171,29 +171,17 @@ export class PageControllerBase {
     }
   }
 
-  get next(): PageLink[] {
-    if (!hasNext(this.pageDef)) {
+  get next(): Link[] {
+    const { def, pageDef } = this
+
+    if (!hasNext(pageDef)) {
       return []
     }
 
-    return this.pageDef.next
-      .map((next) => {
-        const { path } = next
-
-        const page = this.model.pages.find((page) => {
-          return path === page.path
-        })
-
-        if (!page) {
-          return undefined
-        }
-
-        return {
-          ...next,
-          page
-        }
-      })
-      .filter((v) => !!v)
+    // Remove stale links
+    return pageDef.next.filter(({ path }) =>
+      def.pages.some((page) => path === page.path)
+    )
```

Comment:
Rather than loop instantiated `model.pages` we can use `def.pages` instead (lighter)